dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/FourEyeApproverActionlet.java

```diff
@@ -0,0 +1,166 @@
+ * (i.e., all the users assigned to those roles) which will be in charge of approving the new
+ * content. If the users that approve the content are greater or equal to the specified number of
+ * minimum approvers, then the content will move on to the next actionlet in the workflow.
+ * Otherwise, an email will be sent to all users that haven't approved the content yet.
+ *
+ * @author Jose Castro
+ * @version 4.3.0
+ * @since Jan 3, 2018
+ */
+public class FourEyeApproverActionlet extends WorkFlowActionlet {
+
+    private static final long serialVersionUID = 1177885642438262884L;
+
+    private static final String ACTIONLET_NAME = "'4 Eye' Approval";
+    private static final String HOW_TO =
+            "This actionlet implements the '4 Eyes' principle for verifying new content. It takes a comma-separated list of user IDs, email addresses, or role keys "
+                    + "(i.e., the users assigned to those roles) which will be in charge of reviewing the content, and a minimum number of approvers. "
+                    + "If the number of approvers is greater or equal than the specified value, the next sub-action will be executed.";
+    private static final String USER_ID_DELIMITER = ",";
+    private static final String PARAM_CONTENT_APPROVERS = "approvers";
+    private static final String PARAM_MINIMUM_CONTENT_APPROVERS = "minimumApprovers";
+    private static final String PARAM_EMAIL_SUBJECT = "emailSubject";
+    private static final String PARAM_EMAIL_BODY = "emailBody";
+    private static final String PARAM_IS_HTML = "isHtml";
+
+    private static final int DEFAULT_MINIMUM_CONTENT_APPROVERS = 2;
+
+    private boolean shouldStop = false;
+
+    private static ArrayList<WorkflowActionletParameter> ACTIONLET_PARAMETERS = null;
+
+    @Override
+    public synchronized List<WorkflowActionletParameter> getParameters() {
+        if (null == ACTIONLET_PARAMETERS) {
+            ACTIONLET_PARAMETERS = new ArrayList<>();
+            ACTIONLET_PARAMETERS
+                    .add(new MultiUserReferenceParameter(PARAM_CONTENT_APPROVERS,
+                            "User IDs, Emails, or Role Keys", null,
+                            true));
+            ACTIONLET_PARAMETERS
+                    .add(new WorkflowActionletParameter(PARAM_MINIMUM_CONTENT_APPROVERS,
+                            "Number of Approvers",
+                            String.valueOf(DEFAULT_MINIMUM_CONTENT_APPROVERS), true));
+            ACTIONLET_PARAMETERS
+                    .add(new WorkflowActionletParameter(PARAM_EMAIL_SUBJECT,
+                            "Email Subject",
+                            "'4 Eye' Approval Required", true));
+            ACTIONLET_PARAMETERS
+                    .add(new WorkflowActionletParameter(PARAM_EMAIL_BODY, "Email Message",
+                            null,
+                            false));
+        }
+        return ACTIONLET_PARAMETERS;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Return a copy of "ACTIONLET_PARAMETERS". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2384)

Reply:
Not sure why is this syncr, I would create a static code to init it and return the static value every time

Reply:
@jdotcms I think the intention was to do lazy init. But what you said makes sense to me. Also the ACTIONLET_PARAMETERS field should final, at least that what the name in uppercase implies (constant)